versioning/versioning.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+
+GIT_VERSION_COMMAND='git describe --tags --abbrev=8 --dirty 2> /dev/null'
+# Based on https://semver.org/#semantic-versioning-200 but allows `v` in front to match git tags
+GIT_SEMVER_REGEX=/^v?(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
```

Comment:
Should the `v` at the beginning be optional though? The code later strips it to turn it into semver, but should tags without leading `v` be accepted as valid semver tags? It is most like human error because you wouldn't want to mix prefixed and non-prefixed tags inside the same repo.

Reply:
I thought this version script should work with as many projects as possible, so if some decide to not use a `v` in front it should still work I suppose. Let's see what @viovanov thinks about it, I do not mind either way.

Reply:
I think we should always have the `v` in tags. Less decisions to make in projects.
Kubernetes, the linux kernel, CF CLI, cf-deployment have the `v`.
Stratos and docker don't for example, but I think all our projects should have it.

Reply:
I would agree that we should have a consistent version prefix usage in our projects in the long run but we should not enforce it, especially not now since even our own projects differ.